pkg/rtcp/receiver_report.go

```diff
@@ -125,7 +125,7 @@ func (r *ReceiverReport) Unmarshal(rawPacket []byte) error {
 
 	r.SSRC = binary.BigEndian.Uint32(rawPacket[rrSSRCOffset:])
 
-	for i := rrReportOffset; i < len(rawPacket); i += receptionReportLength {
+	for i := rrReportOffset; i < len(rawPacket) && len(r.Reports) < int(h.Count); i += receptionReportLength {
```

Comment:
this works around the fact that extra data at the end of the receiver report is allowed per the standard (RFC 3550) in the "profile-specific-extensions" field. this line causes the parser to ignore the extra data rather then assume it's another reception report (which was causing an error in my application -- chrome was adding data to the end of the packet)

I guess this could be tested by adding data to the end of the 'valid' case in receiver_report_test.go, or should there be a 'valid-with-profile-specific-extension-data' case?

Reply:
I think doing a separate test case for the extension case would be more clear.